### PR TITLE
Change logging system to use custom logger

### PR DESCRIPTION
**Pull request by vmussa**
As suggested by @soukyomi [here](https://github.com/vmussa/qualitube/issues/2), this PR changes the logging system so that it uses a custom logger, enabling user control over format and visibility level.

### CODE REVIEW

File: qualitube/log.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import logging
+import sys
+
+logger = logging.getLogger('qualitube')
+logger.setLevel('INFO')
+logger.addHandler(logging.FileHandler('pipeline.log'))
+logger.addHandler(logging.StreamHandler(sys.stdout))
```
Comment (bitterteriyaki) [inline qualitube/log.py]: The logging handlers must be left for the user to handle. Otherwise unwanted messages will continue to appear on the terminal. If you want messages to appear only when the user uses the CLI app, handlers should only be added in the `qualitube.main.main()` function.

File: qualitube/log.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import logging
+import sys
+
+logger = logging.getLogger('qualitube')
+logger.setLevel('INFO')
```
Comment (bitterteriyaki) [inline qualitube/log.py]: The logging level should be left to the user to define.

File: qualitube/main.py [hunk 3]
```diff
@@ -20,15 +20,8 @@ def chunks(lst: list, n) -> list:
...
-        level=logging.INFO,
-        handlers=[
-            logging.FileHandler('pipeline.log'),
-            logging.StreamHandler(sys.stdout)
-        ]
-    )
-
-    logging.info('Beginning of pipeline.')
+    """Main function."""
+    logger.info('Beginning of pipeline.')
```
Comment (bitterteriyaki) [inline qualitube/main.py]: Here handlers must be added since you want logging messages to appear only in the CLI app:
```suggestion
    """Main function."""
    logger.info('Beginning of pipeline.')
    
    logger.addHandler(logging.FileHandler('pipeline.log'))
    logger.addHandler(logging.StreamHandler(sys.stdout))
```

**Review by bitterteriyaki — CHANGES_REQUESTED** [review]
As it is, logging messages still show up whenever the user uses Qualitube as a library and not as a CLI app. In libraries, it is up to the user to decide how logging messages should be written (from basic terminal logging to ultra-complex file logs). In the case of a CLI app, the library itself can decide how logs will be displayed.

If the user uses Qualitube as a library and decides to setup the logging, he can do it more flexibly:
```py
import logging
import qualitube

# Imagine that I have created my own custom handler,
# with my own formatters and so on:
class MyCustomHandler(logging.StreamHandler):
    ...

# This will get the logger that Qualitube has already created:
logger = logging.getLogger('qualitube')
# Attach Qualitube logger with my custom handler:
logger.addHandler(MyCustomHandler())
```
And of course they can use context managers if they want.

And this way I can use the Qualitube logger as I want. Of course this is a more complex example, a simple logger would be just that:
```py
import logging
import qualitube

logging.basicConfig(level=logging.INFO)
```

In short, what I'm trying to say is to remove the handlers from the `log.py` file and add them inside the `qualitube.main.main()` function. And once again, thanks for the amazing work.

**Comment by vmussa** [comment]
@soukyomi thank you very much for the explanation. Now I think it is done: the handlers are only set in `qualitube.main.main`. As soon as you see and approve this last version I will merge the PR into the main branch and patch it to PyPI. As you can see, I'm not very experienced with logging or developing libraries at all and your help is making me learn a lot, so thanks again!

File: qualitube/main.py (hunk 3 above)
Comment (vmussa) [reply qualitube/main.py]: I didn't see you had already done a suggestion with the hotfix, sorry! I ended up isolating the logger setup in a function.

**Review by bitterteriyaki — APPROVED** [review]
LGTM, TYVM!

**Comment by bitterteriyaki** [comment]
> @soukyomi thank you very much for the explanation. Now I think it is done: the handlers are only set in `qualitube.main.main`. As soon as you see and approve this last version I will merge the PR into the main branch and patch it to PyPI. As you can see, I'm not very experienced with logging or developing libraries at all and your help is making me learn a lot, so thanks again!

Thanks for your dedication! We are here to learn. Any questions about Python loggers, please, contact me!

**Comment by vmussa** [comment]
Cool, @soukyomi! `qualitube v0.1.3` is already available at PyPI!
